extensions/pdh/PerformanceDataMonitor.cpp

```diff
@@ -102,25 +108,36 @@ void PerformanceDataMonitor::onTrigger(core::ProcessContext* context, core::Proc
   rapidjson::Value& body = prepareJSONBody(root);
   for (auto& counter : resource_consumption_counters_) {
     if (counter->collectData())
-      counter->addToJson(body, root.GetAllocator());
+      counter->addToJson(body, root.GetAllocator(), double_precision_);
+  }
+  if (pretty_output_) {
+    utils::PrettyJsonOutputCallback callback(std::move(root));
+    session->write(flowFile, &callback);
+    session->transfer(flowFile, Success);
+  } else {
+    utils::JsonOutputCallback callback(std::move(root));
+    session->write(flowFile, &callback);
+    session->transfer(flowFile, Success);
   }
-  utils::JsonOutputCallback callback(std::move(root));
-  session->write(flowFile, &callback);
-  session->transfer(flowFile, Success);
 }
 
 void PerformanceDataMonitor::initialize() {
-  setSupportedProperties({ CustomPDHCounters, PredefinedGroups, OutputFormatProperty });
+  setSupportedProperties({ CustomPDHCounters, PredefinedGroups, OutputFormatProperty, DoublePrecisionProperty });
```

Comment:
I don't think we usually suffix properties with `Property`

Reply:
changed to DecimalPlaces in [26802f7](https://github.com/martinzink/nifi-minifi-cpp/commit/26802f77ef539757fcedbffea0f596b2ef0cc828)

Reply:
the same thing might apply to `OutputFormatProperty`

Reply:
that conflicts with the PerformanceDataMonitor::OutputFormat enum class

Reply:
it makes sense then